src/connector/src/sink/mod.rs

```diff
@@ -152,13 +206,14 @@ pub struct SinkWriterParam {
     pub meta_client: Option<MetaClient>,
 }
 
-#[async_trait]
-pub trait Sink {
+pub trait Sink: TryFrom<SinkParam, Error = SinkError> {
```

Comment:
```suggestion
pub trait Sink: TryFrom<SinkContext, Error = SinkError> {
```

I would suggest to add sth like `SinkContext`, we may need to add more info such as actor id for better observability.

Reply:
The `Sink` is a set of logical metadata, and it is used in all env such an CN and meta. The compute runtime information, such as actor id and executor id, should only be passed to the sink writer, which is already passed in the `SinkWriterParam` when we create a new sink writer.